src/lib/message/L1ToL2Message.ts

```diff
@@ -259,6 +283,68 @@ export class L1ToL2MessageReader extends L1ToL2Message {
     return null
   }
 
+  private async getFirstRedeem(
+    retryableCreationReceipt?: TransactionReceipt,
+    includeFail?: boolean
+  ): Promise<TransactionReceipt | null> {
+    includeFail = includeFail || false
+    const l2Network = await getL2Network(this.l2Provider)
+    const creationReceipt =
+      retryableCreationReceipt ||
+      (await this.l2Provider.getTransactionReceipt(this.retryableCreationId))
+    const autoRedeem = await this.getAutoRedeem(creationReceipt)
+    if (autoRedeem && (includeFail || autoRedeem.status === 1)) return autoRedeem
+    if (creationReceipt){
+      const creationBlockNumber = creationReceipt.blockNumber
+      const ticketId = creationReceipt.transactionHash
+      const iFace = new Interface([
+        'event RedeemScheduled(     bytes32 indexed ticketId,     bytes32 indexed retryTxHash,     uint64 indexed sequenceNum,     uint64 donatedGas,     address gasDonor )',
+      ])
+      const redeemTopic = iFace.getEventTopic('RedeemScheduled')
+
+      let redeemEventLogs: ethers.providers.Log[] = []
+      let fromBlockNumber: number = creationBlockNumber
```

Comment:
I dont think we need the `fromBlockNumber` variable, could just do `fromBlock.number` where we need that.

Reply:
Ah yeh I see. If you wanted you could detect that by checking that fromBlock.number is the creationBlockNumber eg:
```
await this.l2Provider.getLogs( {fromBlock: fromBlock.number === creationBlockNumber ? fromBlock.number : fromBlock.number + 1, toBlock: toBlockNumber, topics: [redeemTopic, ticketId]})
```

Reply:
I think using a separate variable looks cleaner, no?

Reply:
Yeh, could be a clearer, but I think both ways are a bit messy for different reasons. 

For the separate var I was more concerned about maintainability, having two variables feels like it's potentially bug prone - ie if fromBlock and fromBlockNumber get out of sync with each other for some reason.

Reply:
resolved in 6d914f3a4
turns out we can use `fromBlock.number + 1` for both case as `fromBlock.number` is covered by getAutoRedeem shortcut